server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/CreateMissingParentsTask.java

```diff
@@ -0,0 +1,207 @@
+        }
+
+        private final Instant timestamp;
+        private final Set<String> created;
+        private final long totalCreated;
+        private final Set<String> failures;
+        private final long totalFailure;
+
+        public AdditionalInformation(Instant timestamp, Set<String> created, long totalCreated, Set<String> failures, long totalFailure) {
+            this.created = created;
+            this.failures = failures;
+            this.totalCreated = totalCreated;
+            this.totalFailure = totalFailure;
+            this.timestamp = timestamp;
+        }
+
+        public Set<String> getCreated() {
+            return created;
+        }
+
+        public long getTotalCreated() {
+            return totalCreated;
+        }
+
+        public Set<String> getFailures() {
+            return failures;
+        }
+
+        public long getTotalFailure() {
+            return totalFailure;
+        }
+
+        @Override
+        public Instant timestamp() {
+            return timestamp;
+        }
+    }
+
+    public static class CreateMissingParentsTaskDTO implements TaskDTO {
+        private final String type;
+
+        public CreateMissingParentsTaskDTO(@JsonProperty("type") String type) {
+            this.type = type;
+        }
+
+        @Override
+        public String getType() {
+            return type;
+        }
+    }
+
+    public static TaskDTOModule<CreateMissingParentsTask, CreateMissingParentsTask.CreateMissingParentsTaskDTO> module(MailboxManager mailboxManager) {
+        return DTOModule
+            .forDomainObject(CreateMissingParentsTask.class)
+            .convertToDTO(CreateMissingParentsTask.CreateMissingParentsTaskDTO.class)
+            .toDomainObjectConverter(dto -> new CreateMissingParentsTask(mailboxManager))
+            .toDTOConverter((task, type) -> new CreateMissingParentsTask.CreateMissingParentsTaskDTO(type))
+            .typeName(TASK_TYPE.asString())
+            .withFactory(TaskDTOModule::new);
+    }
+
+    private final ConcurrentLinkedQueue<MailboxId> created;
+    private final AtomicLong totalCreated;
+    private final ConcurrentLinkedQueue<String> failures;
+    private final AtomicLong totalFailure;
+    private final MailboxManager mailboxManager;
+
+    public CreateMissingParentsTask(MailboxManager mailboxManager) {
+        this.mailboxManager = mailboxManager;
+        this.created = new ConcurrentLinkedQueue<>();
+        this.totalCreated = new AtomicLong(0L);
+        this.failures = new ConcurrentLinkedQueue<>();
+        this.totalFailure = new AtomicLong(0L);
+    }
+
+    @Override
+    public Result run() {
+        MailboxSession session = mailboxManager.createSystemSession(USERNAME);
+        try {
+            List<MailboxPath> mailboxPaths = mailboxManager.list(session);
+
+            Set<MailboxPath> parentPaths = mailboxManager.list(session)
```

Comment:
Why don't we reuse mailboxPaths here to save an expensive duplicated read to the DB?